### ISSUE TITLE
`extra_unused_lifetimes` does not warn on spurious `for<'a>`

### ISSUE THREAD

**Issue by jrose-signal** [rust-lang/rust-clippy#13812]
I had the following code

```rust
pub async fn run<F>(body: impl for<'a> FnOnce(PathBuf) -> F)
where
    F: Future<Output = ()>,
{
  unimplemented!()
}
```

A colleague correctly noted I forgot to remove the lifetime qualification after changing from Path to PathBuf (though I probably didn't even need it before, I was experimenting with things). Clippy could have told me that too, but it didn't. (Nor does it for the longer-standing syntax of `&mut dyn for<'a> FnMut(PathBuf)`.)

I *think* `extra_unused_lifetimes` is the lint that should have gotten it, but maybe not.

**Comment by samueltardieu** [comment]
Indeed you should have been warned one way or another, but I wonder whether this is Clippy's job or if it belongs to the compiler proper. Unused generic parameters are not allowed in a type declaration, what about unused HRTB?
